vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -616,10 +616,11 @@ public Receiver getReceiver() {
      */
     public void setReceiver(Receiver receiver) {
         this.receiver = receiver;
-        if (!(receiver instanceof MultiFileReceiver)) {
-            setMaxFiles(1);
+        if (receiver instanceof MultiFileReceiver) {
+            getElement().removeProperty("maxFiles");
+            getElement().executeJs("this.maxFiles = Infinity");
```

Comment:
Should we rather change the default value for this property?

Reply:
We discussed this internally and decided to think about changing the web component's default value later.